realm-annotations-processor/src/main/java/io/realm/processor/ClassMetaData.java

```diff
@@ -307,6 +307,27 @@ private boolean categorizeClassElements() {
                     }
                 }
 
+                if (variableElement.getAnnotation(Required.class) == null) {
+                    // The field doesn't have the @Required annotation
+                    if (!Utils.isPrimitiveType(variableElement) &&
+                            !Utils.isRealmList(variableElement)) {
+                        nullableElements.add(variableElement);
+                    }
+                } else {
+                    // The field has the @Required annotation
+                    if (Utils.isPrimitiveType(variableElement)) {
+                        Utils.error("@Required is not needed for primitive fields - got " + element);
```

Comment:
What is the output of `element.toString()`? Will it make sense to the user?
Ideally this should be something along the lines of `@Required is not needed for primitive fields. Annotated field foo has the type bar`